src/llvm/tools/lld/ELF/Arch/MSP430.cpp

```diff
@@ -24,22 +24,58 @@ using namespace lld::elf;
 namespace {
 class MSP430 final : public TargetInfo {
 public:
+  MSP430();
   RelExpr getRelExpr(RelType Type, const Symbol &S,
                      const uint8_t *Loc) const override;
   void relocateOne(uint8_t *Loc, RelType Type, uint64_t Val) const override;
 };
 } // namespace
 
+MSP430::MSP430() {
+  TrapInstr = 0x4343;
+}
+
 RelExpr MSP430::getRelExpr(RelType Type, const Symbol &S,
                            const uint8_t *Loc) const {
-  return R_ABS;
+  switch (Type) {
+  case R_MSP430_10_PCREL:
+  case R_MSP430_16_PCREL:
+  case R_MSP430_16_PCREL_BYTE:
+  case R_MSP430_2X_PCREL:
+  case R_MSP430_RL_PCREL:
+    return R_PC;
+  default:
+    return R_ABS;
+  }
 }
 
 void MSP430::relocateOne(uint8_t *Loc, RelType Type, uint64_t Val) const {
   switch (Type) {
+  case R_MSP430_8:
+    checkIntUInt(Loc, Val, 8, Type);
+    *Loc = Val;
+    break;
+  case R_MSP430_16:
+  case R_MSP430_16_PCREL:
+    if (Val & 1)
```

Comment:
This line produces the warning: `warning: this statement may fall through [-Wimplicit-fallthrough=]`

Reply:
It should be marked as LLVM_FALLTHROUGH